x/auth/vesting/types/vesting_account.go

```diff
@@ -162,6 +166,38 @@ func (bva BaseVestingAccount) Validate() error {
 	return bva.BaseAccount.Validate()
 }
 
+// UpdateSchedule updates the vesting schedule for a base vesting account.
+// It calculates the proportion of the passed amount that should be used for updating
+// based on the ratio of delegated vesting to total delegation.
+func (bva *BaseVestingAccount) UpdateSchedule(amount sdk.Coins) error {
+	totalDelegated := bva.DelegatedFree.Add(bva.DelegatedVesting...)
+	if totalDelegated.IsZero() {
+		return nil // No delegations, nothing to update
+	}
+
+	// Calculate what portion of the amount should be applied based on delegated vesting ratio
```

Comment:
"should be applied" is vague

```suggestion
	// Calculate the amount to add to the original vesting account based on the delegated vesting ratio.
```